core/experiments/manager.php

```diff
@@ -61,26 +62,51 @@ public function add_feature( array $options ) {
 			'description' => '',
 			'release_status' => self::RELEASE_STATUS_ALPHA,
 			'default' => self::STATE_INACTIVE,
+			'new_site' => [
+				'default_active' => false,
+				'always_active' => false,
+				'minimum_installation_version' => null,
+			],
 			'on_state_change' => null,
 		];
 
-		$allowed_options = [ 'name', 'title', 'description', 'release_status', 'default', 'on_state_change' ];
+		$allowed_options = [ 'name', 'title', 'description', 'release_status', 'default', 'new_site', 'on_state_change' ];
 
 		$experimental_data = $this->merge_properties( $default_experimental_data, $options, $allowed_options );
 
-		$experimental_data = array_merge( $default_experimental_data, $experimental_data );
+		$new_site = $experimental_data['new_site'];
 
-		$state = $this->get_saved_feature_state( $options['name'] );
+		$feature_is_mutable = true;
 
-		if ( ! $state ) {
-			$state = self::STATE_DEFAULT;
+		if ( $new_site['default_active'] || $new_site['always_active'] ) {
+			$is_new_installation = Upgrade_Manager::install_compare( $new_site['minimum_installation_version'], '>=' );
+
+			if ( $is_new_installation ) {
+				if ( $new_site['always_active'] ) {
+					$experimental_data['state'] = self::STATE_ACTIVE;
+
+					$feature_is_mutable = false;
+				} elseif ( $new_site['default_active'] ) {
+					$experimental_data['default'] = self::STATE_ACTIVE;
+				}
+			}
 		}
 
-		$experimental_data['state'] = $state;
+		$experimental_data['mutable'] = $feature_is_mutable;
```

Comment:
Please extract the `new-site` logic to a separate method.

Reply:
It's problematic because some outside variables and the experimental data has to be changed according to different logic results

Reply:
For now, you can add a new state: `STATE_MUTABLE` and the `new_site` will look like:
```
'new_site' => [
	'default' => self::STATE_ACTIVE || self::STATE_MUTABLE,
	'minimum_installation_version' => null,
],
```

Reply:
Mutable is not a state and it can't be set or changed. It's only an internal setting to determine whether the feature state can be edited by the user